ui/src/app/views/cohort-builder/search/cohort-builder/cohort-builder.component.ts

```diff
@@ -0,0 +1,205 @@
+import { Component, OnInit, ChangeDetectorRef, OnDestroy, Inject } from '@angular/core';
+import { Router } from '@angular/router';
+import {DOCUMENT} from '@angular/platform-browser';
+import { SearchGroup, Subject, SearchResult } from '../model';
+import 'rxjs/add/operator/takeWhile';
+import { BroadcastService } from '../service/broadcast.service';
+import { intersection, complement, union } from 'set-manipulator';
+
+@Component({
+  selector: 'app-search',
+  templateUrl: 'cohort-builder.component.html',
+  styleUrls: ['cohort-builder.component.css']
+})
+export class CohortBuilderComponent implements OnInit, OnDestroy {
+
+  /**
+   * The search groups that make up the inclusion group.
+   *
+   * @public
+   * @type {SearchGroup[]}
+   * @memberof CohortBuilderComponent
+   */
+  public searchGroups: SearchGroup[];
+
+  /**
+   * The search groups that make up the exclusion group.
+   *
+   * @public
+   * @type {SearchGroup[]}
+   * @memberof CohortBuilderComponent
+   */
+  public exclusionGroups: SearchGroup[];
+
+  private alive = true;
+
+  totalSet: Subject[] = [];
+
+  constructor(private changeDetectorRef: ChangeDetectorRef,
+              private broadcastService: BroadcastService,
+              private router: Router,
+              @Inject(DOCUMENT) private document: any) {}
+
+  ngOnInit() {
+    this.searchGroups = [new SearchGroup()];
+    this.exclusionGroups = [new SearchGroup('Exclude participants where:')];
+
+    this.broadcastService.updatedCounts$
+      .takeWhile(() => this.alive)
+      .subscribe(change => {
+        this.updateGroupSet(change.searchGroup, change.searchResult);
+        this.updateTotalSet();
+        this.updateCharts();
+      });
+    this.broadcastService.removedSearchResult$
+      .takeWhile(() => this.alive)
+      .subscribe(change => {
+        this.updateGroupSetRemovedSearchResult(change.searchGroup);
+        this.updateTotalSet();
+        this.updateCharts();
+      });
+  }
+
+  addSearchGroup() {
+    this.searchGroups.push(new SearchGroup());
+    this.changeDetectorRef.detectChanges();
+    const scrollableDiv = window.document.getElementById('scrollable-groups');
```

Comment:
Is it possible to use https://stackoverflow.com/a/35209681 or a similar technique to get a hold of this element @freemabd1 ?